lib/Dialect/Triton/Transforms/Combine.cpp

```diff
@@ -49,6 +49,14 @@ DenseElementsAttr getConstantValue(Builder &builder, Attribute value,
   return res;
 }
 
+bool isAddPtrOffsetCombinable(Value first, Value second) {
+  if (first.getType().isa<ShapedType>() && second.getType().isa<ShapedType>())
+    return first.getType().cast<ShapedType>().getElementType() ==
+           second.getType().cast<ShapedType>().getElementType();
```

Comment:
in what case would `first.getType().cast<ShapedType>().getElementType() == second.getType().cast<ShapedType>().getElementType();` be true but `first.getType() == second.getType();` would not? Isn't `first.getType() == second.getType();` always the right check?

Reply:
since the type of offset is TT_IntLike whihc are AnyTypeOf<[TT_Int, TT_IntTensor]>, the code here checks both type of TT_Int and element type of tensor. 
Also addptr only constrain on shape, so when it comes with tensor, the element type may mismath. like:
```

    %ptr0 = tt.addptr %base_, %idx0 : tensor<8x!tt.ptr<f32>>, tensor<8xi16>
    %ptr1 = tt.addptr %ptr0, %idx1 : tensor<8x!tt.ptr<f32>>, tensor<8xi32>

```

Reply:
in this case checking `first.getType() == second.getType()` still seems like the right check, this would return `false` in your example which is what we want.